api/src/main/java/org/openmrs/module/registrationcore/api/mpi/pixpdq/HttpSender.java

```diff
@@ -0,0 +1,98 @@
+package org.openmrs.module.registrationcore.api.mpi.pixpdq;
+
+
+import ca.uhn.hl7v2.HL7Exception;
+import ca.uhn.hl7v2.llp.LLPException;
+import ca.uhn.hl7v2.model.Message;
+import ca.uhn.hl7v2.parser.PipeParser;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.xerces.impl.dv.util.Base64;
+import org.openmrs.module.registrationcore.api.impl.RegistrationCoreProperties;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class HttpSender implements Hl7v2Sender {
+
+    private final Log log = LogFactory.getLog(this.getClass());
+
+    @Autowired
+    private RegistrationCoreProperties config;
+
+    @Override
+    public Message sendPdqMessage(Message request) throws LLPException, IOException, HL7Exception {
+        URL url = new URL(config.getPdqEndpoint());
+        return sendMessage(request, url);
+    }
+
+    @Override
+    public Message sendPixMessage(Message request) throws LLPException, IOException, HL7Exception {
+        URL url = new URL(config.getPixEndpoint());
+        return sendMessage(request, url);
+    }
+
+    private Message sendMessage(Message request, URL url) throws HL7Exception, LLPException, IOException {
+
+        PipeParser parser = new PipeParser();
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod("POST");
+        connection.setRequestProperty("Content-Type", "application/hl7-v2+er7; charset=utf-8");
+        String authentication = config.getMpiUsername() + ":" + config.getMpiPassword();
+        String encoded = "Basic "+ Base64.encode(authentication.getBytes());
+        connection.setRequestProperty("Authorization", encoded);
+        connection.setRequestMethod("POST");
+        connection.setDoOutput(true);
+
+        try
+        {
+            if(log.isDebugEnabled())
+                log.debug(String.format("Sending to %s : %s", url, parser.encode(request)));
```

Comment:
By convention, all statements need to be inside the braces for if clauses

Reply:
refactored to remove this pattern of logging as per https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-HowTo(Not)Log